### PR TITLE
Support for java.util.zip.{ZipFile, ZipEntry}

### PR DESCRIPTION
**Pull request by Duhemm**
This code has been ported from Apache Harmony.

### CODE REVIEW

File: javalib/src/main/scala/java/util/zip/ZipEntry.scala [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+  //   cal.set(1980 + ((modDate >> 9) & 0x7f),
+  //           ((modDate >> 5) & 0xf) - 1,
+  //           modDate & 0x1f,
+  //           (time >> 11) & 0x1f,
+  //           (time >> 5) & 0x3f,
+  //           (time & 0x1f) << 1)
+  //   cal.getTime().getTime()
+  // } else {
+  //   -1
+  // }
```
Comment (densh) [inline javalib/src/main/scala/java/util/zip/ZipEntry.scala]: Can you file an issue for this and setTime? TODOs in code are often overlooked.

File: javalib/src/main/scala/java/util/zip/ZipEntry.scala [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+      }
+
+    try {
+      /*
+       * The actual character set is "IBM Code Page 437".  As of
+       * Sep 2006, the Zip spec (APPNOTE.TXT) supports UTF-8.  When
+       * bit 11 of the GP flags field is set, the file name and
+       * comment fields are UTF-8.
+       *
+       * TODO: add correct UTF-8 support.
```
Comment (densh) [inline javalib/src/main/scala/java/util/zip/ZipEntry.scala]: Needs an issue instead of TODO.

Comment (Duhemm) [inline javalib/src/main/scala/java/util/zip/ZipEntry.scala]: Sure.
Just for the records, this comment comes from Harmony's source code:
https://github.com/apache/harmony/blob/java6/classlib/modules/archive/src/main/java/java/util/zip/ZipEntry.java#L404-L411

Comment (Duhemm) [inline javalib/src/main/scala/java/util/zip/ZipEntry.scala]: Opened #937

File: javalib/src/main/scala/java/util/zip/ZipEntry.scala (hunk 1 above)
Comment (Duhemm) [inline javalib/src/main/scala/java/util/zip/ZipEntry.scala]: Opened #936

**Comment by densh** [comment]
LGTM, nicely done!

**Pull request merged**